apps/web/src/components/Composer/Actions/Attachment.tsx

```diff
@@ -91,99 +90,118 @@ const Attachment: FC = () => {
-              )
-            }
+            <label
+              className={cn(
+                { 'cursor-pointer': !disableImageUpload() },
+                { 'cursor-default': disableImageUpload() },
+                '!flex items-center gap-1 space-x-1'
+              )}
+              htmlFor={`image_${id}`}
+            >
+              <PhotoIcon className="text-brand-500 size-5" />
+              <span className="text-sm">Upload image(s)</span>
+
+              <input
+                accept={ImageMimeType.join(',')}
+                className="hidden"
+                disabled={disableImageUpload()}
+                id={`image_${id}`}
+                multiple
+                onChange={handleAttachment}
+                type="file"
+              />
+            </label>
+          </DropdownMenuPrimitive.Item>
+          <DropdownMenuPrimitive.Item
+            className="rounded-lg menu-item data-[highlighted]:bg-gray-100 dark:data-[highlighted]:bg-gray-800 data-[disabled]:cursor-default"
             disabled={Boolean(attachments.length)}
-            htmlFor={`video_${id}`}
+            onSelect={preventDefaultEvent}
           >
-            <VideoCameraIcon className="text-brand-500 size-4" />
-            <span className="text-sm">Upload video</span>
-            <input
-              accept={VideoMimeType.join(',')}
-              className="hidden"
-              disabled={Boolean(attachments.length)}
-              id={`video_${id}`}
-              onChange={handleAttachment}
-              type="file"
-            />
-          </Menu.Item>
-          <Menu.Item
-            as="label"
-            className={({ active }) =>
-              cn(
-                { 'dropdown-active': active },
-                'menu-item !flex cursor-pointer items-center gap-1 space-x-1 rounded-lg'
-              )
-            }
+            <label
+              className={cn(
+                { 'cursor-pointer': !Boolean(attachments.length) },
+                { 'cursor-default': Boolean(attachments.length) },
+                '!flex items-center gap-1 space-x-1'
+              )}
+              htmlFor={`video_${id}`}
+            >
+              <VideoCameraIcon className="text-brand-500 size-5" />
+              <span className="text-sm">Upload video</span>
+
+              <input
+                accept={VideoMimeType.join(',')}
+                className="hidden"
+                disabled={Boolean(attachments.length)}
+                id={`video_${id}`}
+                onChange={handleAttachment}
+                type="file"
+              />
+            </label>
+          </DropdownMenuPrimitive.Item>
+          <DropdownMenuPrimitive.Item
+            className="rounded-lg menu-item data-[highlighted]:bg-gray-100 dark:data-[highlighted]:bg-gray-800 data-[disabled]:cursor-default"
             disabled={Boolean(attachments.length)}
-            htmlFor={`audio_${id}`}
+            onSelect={preventDefaultEvent}
```

Comment:
why we need to prevent it?

Reply:
DropdownMenuPrimitive.Item has default handler attached to it because of which the call was not going to its children i.e.  <label>,  to supress this behaviour, used prevent default.